### PR TITLE
(6/n) Convert Payment Protocol Buffers

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
Provide a native Java wrapper for the [Payment protocol buffer](https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L224).
Analogue in Swift: https://github.com/xpring-eng/XpringKit/pull/131
Analogue in JS: https://github.com/xpring-eng/Xpring-JS/pull/254

### Context of Change
We want to provide an API which returns transaction history. This API shouldn't return raw protocol buffers as they have some slightly complex abstractions and because that means we're tying a network layer to a public API.

This is the start of building up several types to ultimately serve all transaction types.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After
No change.

## Test Plan
New CI introduced for unit tests.

## Future Tasks

7 - 9 / 9 will build up to Transaction protobuf conversion

### CODE REVIEW

File: src/main/java/io/xpring/xrpl/model/XRPPayment.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * @return (Optional) Highest amount of source currency this transaction is allowed to cost.
+     */
+    @Nullable XRPCurrencyAmount sendMax();
+
+    static XRPPayment from(Payment payment) {
+        // amount is required
+        XRPCurrencyAmount amount = XRPCurrencyAmount.from(payment.getAmount().getValue());
+        if (amount == null) { return null; }
```
Comment (keefertaylor) [inline src/main/java/io/xpring/xrpl/model/XRPPayment.java]: You can shorten to 

```
if (!amount) { ... } 
```

File: src/main/java/io/xpring/xrpl/model/XRPPayment.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                .amount(amount)
+                .destination(destination)
+                .destinationTag(destinationTag)
+                .deliverMin(deliverMin)
+                .invoiceID(invoiceID)
+                .paths(paths)
+                .sendMax(sendMax)
+                .build();
+    }
+}
```
Comment (keefertaylor) [inline src/main/java/io/xpring/xrpl/model/XRPPayment.java]: nit: newline

File: src/main/java/io/xpring/xrpl/model/XRPPayment.java (hunk 1 above)
Comment (amiecorso) [reply src/main/java/io/xpring/xrpl/model/XRPPayment.java]: This is also giving me an issue about requiring a boolean when I try to write it like that...

File: src/main/java/io/xpring/xrpl/model/XRPPayment.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            destinationTag = payment.getDestinationTag().getValue();
+        } else {
+            destinationTag = null;
+        }
+
+        // If the deliverMin field is set, it must be able to be transformed into an XRPCurrencyAmount.
+        XRPCurrencyAmount deliverMin;
+        if (payment.hasDeliverMin()) {
+            deliverMin = XRPCurrencyAmount.from(payment.getDeliverMin().getValue());
+            if (deliverMin == null) { return null; }
```
Comment (nkramer44) [inline src/main/java/io/xpring/xrpl/model/XRPPayment.java]: nit: Separate the conditional to different lines

File: src/main/java/io/xpring/xrpl/model/XRPPayment.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                                     .collect(Collectors.toList());
+        if (paths.isEmpty()) {
+            paths = null;
+        }
+
+        // If the sendMax field is set, it must be able to be transformed into an XRPCurrencyAmount.
+        XRPCurrencyAmount sendMax;
+        if (payment.hasSendMax()) {
+            sendMax = XRPCurrencyAmount.from(payment.getSendMax().getValue());
+            if (sendMax == null) { return null; }
```
Comment (nkramer44) [inline src/main/java/io/xpring/xrpl/model/XRPPayment.java]: same here

**Comment by nkramer44** [comment]
Woops, was a little late to the party

**Pull request merged**